### PR TITLE
Attachments manager - Pagination

### PR DESCRIPTION
**Pull request by Narcolapser**
Added pagination for when there are a lot of attachments.

### CODE REVIEW

**Comment by jonathanmtran** [comment]
Instead of building the portlet URL manually, have you considered using `<portlet:renderURL></portlet:renderURL>` in the view?

See http://www.opensource-techblog.com/2014/12/create-render-url-by-portlet-tag-in-jsp.html

**Comment by drewwills** [comment]
The point made by @jonathanmtran is a good one -- only the portal framework Java code (according to the spec) is allowed to put characters in a sequence that creates a portlet URL.

**Comment by Narcolapser** [comment]
There are a lot of tag libraries that I don't use because I don't know what they are. I certainly like this approach better since it skips the awkward fname issue. Thanks for the tip.

File: src/main/webapp/WEB-INF/jsp/attachments-manager/view.jsp [hunk 1]
```diff
@@ -126,6 +126,13 @@ div.col_content{
 }
 </style>
 
+<portlet:renderURL var="prevPage">
+    <portlet:param name="page" value="${page-1}"/>
```
[the remark this answers is not in the document]
Comment (Narcolapser) [reply src/main/webapp/WEB-INF/jsp/attachments-manager/view.jsp]: line 148 has a conditional for that.

**Comment by drewwills** [comment]
@Narcolapser -- are you able to help with this issue?...

  - https://github.com/Jasig/SimpleContentPortlet/issues/42

We have someone at Unicon looking at it, but you may be able to help based on your knowledge of FineUploader.

**Comment by ChristianMurphy** [comment]
@Narcolapser @drewwills @jonathanmtran any remaining items for this?

**Comment by Narcolapser** [comment]
Technically no. There is the regression that was introduced in #42, but that is technically separate. I've looked into it a little, but I actually know very little about FineUploader or the other portlets that make use of the attachments portlet.

**Comment by Narcolapser** [comment]
#43 fixed #42, which was technically never a problem for this pull request. Not that I think it really matters. I'm closing this pull request as I, the writer of it, do not think this management portlet is particularly useful. I ended up creating an entirely separate portlet to manage the portal's attachments. I'd share but it uses hardcoded tie ins to make this simple. I skip Hibernate entirely and go straight to SQL as well, further compounding the portability issue. Overall I'd say the SimpleContentPortlet's attachment subsystem was not designed to be a file manager and trying to shoehorn that functionality in did not result in a great experience. This portlet, even with pagination, will crash and respond super slowly once you get above ~2-3gb of attachments.

**Pull request closed without merging**